### ISSUE TITLE
solid/reactivity: setup-level signal read captured in a returned closure lints clean, while the correct accessor fix is flagged

### ISSUE THREAD

**Issue by ryansolid** [joshwilsonvu/eslint-plugin-solid#213]
**Version:** eslint-plugin-solid 0.16.0 (`configs/v2`, run via oxlint 1.79 jsPlugins — same result as under eslint), solid-js 2.0.0-rc.3.

`solid/reactivity` gets this pair backwards — a false negative on a real reactivity bug and a warning on its correct fix:

```ts
import { createSignal } from 'solid-js';
const [items] = createSignal([1, 2, 3]);

// BUG — no warning. `items()` is read once at setup (untracked), and the
// returned accessor closes over the frozen result. Consumers calling
// `total()` in JSX/effects never update. This is the classic
// STRICT_READ_UNTRACKED footgun Solid 2's dev mode flags at runtime.
export function useCartTotalBuggy() {
  const list = items();
  return () => list.reduce((sum, n) => sum + n, 0);
}

// CORRECT — warned: "This function should be passed to a tracked scope
// (like createEffect) or an event handler because it contains reactivity,
// or else changes will be ignored."
export function useCartTotalFixed() {
  return () => items().reduce((sum, n) => sum + n, 0);
}
```

Returning an accessor from a hook-style function is the idiomatic Solid pattern, so the warning on the fixed version reads as a false positive in practice (understood that the rule can't prove the caller tracks it — but the same uncertainty exists for any custom primitive returning an accessor, including what `createMemo` itself returns).

The combination is what makes it damaging: someone (or an AI agent) who leans on the linter is warned *away* from the correct minimal fix while the actual defect lints clean. We hit this in an agent evaluation on the solid-v2 templates, where an agent's correct fix triggered the warning and it had to trust runtime diagnostics evidence over the lint output; switching to `createMemo` silenced the rule, but the minimal accessor fix was behaviorally equivalent and idiomatic.

Two suggestions, independently useful:

1. **Catch the false negative** (the more harmful half): a signal/props read at the top level of a function body whose result is captured by a *returned* function is almost always this bug — the value is frozen while the returned closure pretends to be live. That's a narrower, more confident pattern than general untracked-read detection.
2. **Reconsider the warning for returned accessors**: when the read is a direct signal call inside a plain closure that the function *returns* (hook-style), consider downgrading or exempting — the return value's contract is "accessor", and the current warning pushes people toward wrapping in `createMemo` purely to appease the rule.

Happy to provide more context from the evaluation if useful.

**Comment by ryansolid** [comment]
Fixed by #214 (merged): setup-level signal reads captured by returned closures now report the new `staleCapture` message, and directly-returned accessors are exempt from the tracked-scope requirement (the custom-primitive contract). Ships in the next release.
